packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -106,6 +106,17 @@ abstract class WidgetsBindingObserver {
   /// [SystemChannels.navigation].
   Future<bool> didPushRoute(String route) => Future<bool>.value(false);
 
+  /// Called when the host tells the app to push a new route information onto
+  /// the router.
+  ///
+  /// Observers are expected to return true if they were able to
+  /// handle the notification. Observers are notified in registration
+  /// order until one returns true.
+  ///
+  /// This method exposes the `pushRouteInformation` notification from
+  /// [SystemChannels.router].
+  Future<bool> didPushRouteInformation(String route, Object state) => Future<bool>.value(false);
```

Comment:
instead, because the web engine will notify both router and navigator channel. These two method need to be two complete separate path. Otherwise, it may double notify the listeners